provider/openai-api/query.go

```diff
@@ -0,0 +1,32 @@
+package openaiapi
+
+import (
+	"context"
+	"fmt"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/sashabaranov/go-openai"
+)
+
+// QueryOpenAIModel queries an OpenAI API model.
+func QueryOpenAIAPIModel(ctx context.Context, client *openai.Client, modelIdentifier string, promptText string) (response string, err error) {
```

Comment:
Wouldn't it be easier to just embed the openAI provider in Ollama/Openrouter?

Reply:
Was thinking about it as well... but the openAI provider i.e. has a models slice an neither Ollama nor Openrouter need that. So... we can embed but it feels a bit mushy.

Reply:
Hmm yeah I know what you mean. Alright then, ignore that one.